x-pack/plugins/ml/public/application/overview/components/anomaly_detection_panel/utils.ts

```diff
@@ -157,11 +162,11 @@ export function getStatsBarData(jobsList: any) {
   return jobStats;
 }
 
-export function getJobsFromGroup(group: Group, jobs: any) {
-  return group.jobIds.map((jobId) => jobs[jobId]).filter((id) => id !== undefined);
+export function getJobsFromGroup(group: Group, jobs: Record<string, any>) {
```

Comment:
is it worth trying to get rid of some of these `any` types.
the function `getJobsWithTimerange` below looks a bit messy. It's returning a brand new type of job object as `any`, but is then being used later as a `MlSummaryJobs` which is completely wrong.
But it doesn't seem to matter because only the id from the object is being used when it's passed to this (`getJobsFromGroup`) function.
It would be good to clean this code up. It looks like `getJobsWithTimerange` isn't really used properly at all, as in, the time range parts are ignored.

Reply:
refactored in 95171b2a377488c53244ec1c4f9e31a38b366c4c